### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by evilnick**
here's all the bits for the readme.

NB 
 - if you want to 'tidy up' the files will need to edit the action 
 - the roadmap stuff will be added in a future PR

### CODE REVIEW

File: .github/workflows/readme.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out repo
+      uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - uses: actions/cache@v2
```
Comment (addyess) [inline .github/workflows/readme.yml]: Can we try 

```suggestion
    - uses: actions/cache@v3
```

File: .github/workflows/readme.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  schedule:
+    - cron: '36 8 * * *'
+    - cron: '36 14 * * *'
+    - cron: '36 21 * * *'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out repo
+      uses: actions/checkout@v2
```
Comment (addyess) [inline .github/workflows/readme.yml]: Let's try:
```suggestion
      uses: actions/checkout@v3
```

File: .github/workflows/readme.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    - name: Set up Python
+      uses: actions/setup-python@v2
```
Comment (addyess) [inline .github/workflows/readme.yml]: What about 
```suggestion
      uses: actions/setup-python@v4
```

File: .github/workflows/readme.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        python -m pip install -r requirements.txt
+    - name: Update README
+      run: |-
+        python dynamic_readme.py
+        cat README.md
+    - name: Commit and push if README changed
+      run: |-
+        git diff
+        git config --global user.email "bot@example.com"
+        git config --global user.name "README-bot"
```
Comment (addyess) [inline .github/workflows/readme.yml]: All the docs I see for doing this recommend this

```suggestion
        git config --global user.name "${{ env.CI_COMMIT_AUTHOR }}"
        git config --global user.email "username@users.noreply.github.com"
```

File: dynamic_readme.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            "title": entry["title"],
+            "url": entry["link"].split("#")[0],
+            "published": entry["published"].split("+")[0],
+        }
+        for entry in entries
+    ]
+
+
+def fetch_activity():
```
Comment (addyess) [inline dynamic_readme.py:30]: this is so dang clever 🤩

**Review by addyess — APPROVED** [review]
I've made suggestions you can take/keep.  but dang this is freaking cool
take all my stars ⭐ ⭐ ⭐ ⭐ ⭐

File: .github/workflows/readme.yml (hunk 4 above)
Comment (evilnick) [reply .github/workflows/readme.yml]: seems fair

Comment (addyess) [reply .github/workflows/readme.yml]: [Ok, i'm not good at reading](https://joht.github.io/johtizen/build/2022/01/20/github-actions-push-into-repository.html)

Please revert my suggestion.  Maybe something like `cdkbot@gmail.com` is better 😬

Comment (addyess) [reply .github/workflows/readme.yml]: > ⓘ Note that the username in username@users.noreply.github.com needs to be replaced.

also, that name should be `cdkbot`

File: .github/workflows/readme.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        python -m pip install -r requirements.txt
+    - name: Update README
+      run: |-
+        python dynamic_readme.py
+        cat README.md
+    - name: Commit and push if README changed
+      run: |-
+        git diff
+        git config --global user.name "${{ env.CI_COMMIT_AUTHOR }}"
+        git config --global user.email "username@users.noreply.github.com"
```
Comment (evilnick) [inline .github/workflows/readme.yml]: @addyess ah...this wants an actual username. D'oh! - that will teach me for just accepting your changes

Comment (addyess) [inline .github/workflows/readme.yml]:
```suggestion
        git config --global user.name "cdkbot"
        git config --global user.email "cdkbot@gmail.com"
```

Comment (addyess) [reply .github/workflows/readme.yml]: yeah i'm so sorry... i just gave you a really crappy suggestion because i'm incapable of reading slowly

Comment (addyess) [reply .github/workflows/readme.yml]: this is a valid actually commit-able suggestion

**Comment by evilnick** [comment]
Hurrah! I'm calling that done then

**Pull request merged**